plugins/inputs/docker/docker.go

```diff
@@ -297,6 +316,25 @@ func (d *Docker) gatherContainer(
 		}
 	}
 
+	inspectCtx, inspectCancel := context.WithTimeout(context.Background(), d.Timeout.Duration)
+	defer inspectCancel()
+	ic, err := inspectWrapper(d.client, inspectCtx, container.ID)
+	if err != nil {
+		return err
+	}
+	// Iterate over the container environment variables and see if there are any variables to be tagged
+	for _, env := range ic.Config.Env {
```

Comment:
Ah you're right. I should iterate over the whitelisted tags as it will be better and just see if they are in the docker container